Code/RDGeneral/Ranking.h

```diff
@@ -72,14 +72,13 @@ void rankVect(const std::vector<T1> &vect, T2 &res) {
   std::sort(indices.begin(), indices.end(), argless<std::vector<T1>>(vect));
 
   int currRank = 0;
-  T1 lastV = vect[indices[0]];
+  unsigned int lastIdx = indices[0];
```

Comment:
Did you consider a version of rankVect with a memory workspace?

    class RankVect {
        template <typename T1, typename T2> void operator(const std::vector<T1> &vect, T2 &res) {...}

    private:
        std::vector<unsigned int> m_indices;
    };

Reply:
I was actually intending to pass indices as a reference in a follow-on change (and the vector would be allocated once at the top of `iterateCIPRanks` leading to effectively the same performance benefit), so the parent could read it to avoid re-processing atoms with unique ranks.

We could accomplish the same by giving access to the `m_indices` member.  Do you have a preference between the two approaches?